File: q4.py
#  Given an array which consists of only 0, 1 and 2. Sort the array without using any sorting algo
#User function Template for python3

def sort012(arr,n):
    # code here
    z,o,t = 0,0,0
    for i in arr:
        if i==0:
            z = z+1
        elif i==1:
            o = o+1
        else:
            t = t+1
    for i in range(n):
        if z:
            arr[i] = 0
            z = z-1
        elif o:
            arr[i] = 1
            o = o-1
        else:
            arr[i] = 2
            t = t-1

#{ 
#  Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t=int(input())
    for _ in range(t):
        n=int(input())
        arr=[int(x) for x in input().strip().split()]
        sort012(arr,n)
        for i in arr:
            print(i, end=' ')
        print()

# } Driver Code Ends

File: q4.cpp
// Given an array which consists of only 0, 1 and 2. Sort the array without using any sorting algo
// { Driver Code Starts
#include<bits/stdc++.h>
using namespace std;
void sort012(int[],int);

int main() {

    int t;
    cin >> t;

    while(t--){
        int n;
        cin >>n;
        int a[n];
        for(int i=0;i<n;i++){
            cin >> a[i];
        }

        sort012(a, n);

        for(int i=0;i<n;i++){
            cout << a[i]  << " ";
        }

        cout << endl;
        
        
    }
    return 0;
}

// } Driver Code Ends


void sort012(int a[], int n)
{
    int zero=0,one=0,two=0,i;
    // coode here 
    for(i=0;i<n;i++)
    {
        if(a[i]==0){
            zero++;
        }
        else if(a[i]==1){
            one++;
        }
        else{
            two++;
        }
    }
    for(i=0;i<n;i++)
    {
        if(zero){
            a[i]=0;
            zero--;
        }
        else if(one){
            a[i]=1;
            one--;
        }
        else{
            a[i]=2;
            two--;
        }
    }
}